clients/src/main/java/org/apache/kafka/common/ApiKey.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common;
+
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+
+/**
+ * Identifiers for Kafka APIs.
+ *
+ * This is a public API for both Kafka clients and internals.  Metadata which is not intended to be
+ * exposed to end-users should go in #{org.apache.kafka.common.protocol.ApiKeys} rather than here.
+ *
+ * Each ApiKey represents an API which Kafka brokers understand.  In order to maintain
+ * compatibility, we can add more APIs to the list, but not remove or reorder existing APIs.
+ */
+public enum ApiKey {
+    PRODUCE("Produce", 0, 3),
+    FETCH("Fetch", 0, 5),
+    LIST_OFFSETS("Offsets", 0, 2),
```

Comment:
It is currently called "Offsets" -- it seems like changing this should be a separate JIRA?